pkg/envoy/admin/kds_client.go

```diff
@@ -163,3 +165,20 @@ func resNameInZone(nameInGlobal string, k8sStore bool) (string, string, error) {
 	}
 	return zone, nameInZone, nil
 }
+
+type KDSTransportError struct {
+	requestType string
+	reason      string
+}
+
+func (e *KDSTransportError) Error() string {
+	if e.reason == "" {
+		return fmt.Sprintf("could not send %s", e.requestType)
+	} else {
+		return fmt.Sprintf("could not send %s: %s", e.requestType, e.reason)
+	}
+}
+
+func (e *KDSTransportError) Is(err error) bool {
+	return reflect.TypeOf(e) == reflect.TypeOf(err)
```

Comment:
Shouldn't this just be like:

```suggestion
	_, ok := target.(*KDSTransportError)
	return ok
```

?

Reply:
All other `Is` implementations are using `reflect`:
```
$ grep -r ") Is(err" .
./pkg/api-server/types/errors.go:func (a *InvalidPageSizeError) Is(err error) bool {
./pkg/core/access/errors.go:func (a *AccessDeniedError) Is(err error) bool {
./pkg/core/resources/store/store.go:func (e *ResourceConflictError) Is(err error) bool {
./pkg/core/resources/store/store.go:func (e *AssertionError) Is(err error) bool {
./pkg/core/resources/store/store.go:func (a *PreconditionError) Is(err error) bool {
./pkg/core/rest/errors/errors.go:func (e *BadRequest) Is(err error) bool {
./pkg/intercp/envoyadmin/forwarding_kds_client.go:func (e *StreamNotConnectedError) Is(err error) bool {

$ grep -r "reflect.TypeOf(.*) == reflect.TypeOf(err)" .
./pkg/api-server/types/errors.go:       return reflect.TypeOf(a) == reflect.TypeOf(err)
./pkg/core/access/errors.go:    return reflect.TypeOf(a) == reflect.TypeOf(err)
./pkg/core/resources/store/store.go:    return reflect.TypeOf(e) == reflect.TypeOf(err)
./pkg/core/resources/store/store.go:    return reflect.TypeOf(e) == reflect.TypeOf(err)
./pkg/core/resources/store/store.go:    return reflect.TypeOf(a) == reflect.TypeOf(err)
./pkg/core/rest/errors/errors.go:       return reflect.TypeOf(e) == reflect.TypeOf(err)
./pkg/intercp/envoyadmin/forwarding_kds_client.go:      return reflect.TypeOf(e) == reflect.TypeOf(err)

```

If we want to change this I can open another PR that updates all the places

Reply:
Is there much a difference?

Reply:
Avoiding reflections (if possible) is a good approach in general. But I don't think we should change it in this PR.